Although the following patch has been addressed in Icehouse 2014.1, it
still has not yet been released for Havana.  It has been modified to
apply cleanly into our current Havana implementation

commit 05ba5f81c390efbb6e5eb98db62f8f2599389ddb
Author: Matt Fischer <matt@mattfischer.com>
Date:   Thu Jan 9 15:35:38 2014 -0700

    Don't reset LOCAL_PATH if it's already set
    
    If LOCAL_PATH is set in local_settings, then settings should
    not override it, even if SECRET_KEY is unset.
    
    Change-Id: I6aca03f65afddffb6cdd00e4084a8ab9b2255ef1
    Closes-Bug: #1254694

diff --git a/openstack_dashboard/settings.py b/openstack_dashboard/settings.py
--- horizon-2013.2.3/openstack_dashboard/settings.py.~1~	2014-04-03 11:45:53.000000000 -0700
+++ horizon-2013.2.3/openstack_dashboard/settings.py	2014-05-25 13:48:46.166574128 -0700
@@ -206,6 +206,7 @@
 }
 
 SECRET_KEY = None
+LOCAL_PATH = None
 
 try:
     from local.local_settings import *  # noqa
@@ -216,9 +217,11 @@
 # file is present. See local_settings.py.example for full documentation on the
 # horizon.utils.secret_key module and its use.
 if not SECRET_KEY:
+    if not LOCAL_PATH:
+        LOCAL_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                                  'local')
+
     from horizon.utils import secret_key
-    LOCAL_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)),
-                              'local')
     SECRET_KEY = secret_key.generate_or_read_from_file(os.path.join(LOCAL_PATH,
                                                        '.secret_key_store'))
 
